package/Runtime/Plugins/iOS/NimbusManager.swift

```diff
@@ -0,0 +1,196 @@
+        guard let viewController = unityViewController() else { return }
+        
+        self.destroyExistingAd() // TODO: confirm if this is the expected
+        
+        let request = NimbusRequest.forInterstitialAd(position: position)
+        request.impressions[0].banner?.bidFloor = bannerFloor
+        request.impressions[0].video?.bidFloor = videoFloor
+        
+        nimbusAdManager = NimbusAdManager()
+        nimbusAdManager?.delegate = self
+        nimbusAdManager?.showRewardedVideoAd(request: request,
+                                             closeButtonDelay: closeButtonDelay,
+                                             adPresentingViewController: viewController)        
+    }
+    
+    @objc public func showRewardedVideoAd(position: String, videoFloor: Float, closeButtonDelay: Double) {
+        guard let viewController = unityViewController() else { return }
+        
+        self.destroyExistingAd() // TODO: confirm if this is the expected
+        
+        let request = NimbusRequest.forRewardedVideo(position: position)
+        request.impressions[0].video?.bidFloor = videoFloor
+        
+        nimbusAdManager = NimbusAdManager()
+        nimbusAdManager?.delegate = self
+        nimbusAdManager?.showRewardedVideoAd(request: request,
+                                             closeButtonDelay: closeButtonDelay,
+                                             adPresentingViewController: viewController)
+    }
+    
+    @objc public func setGDPRConsentString(consent: String) {
+        var user = NimbusRequestManager.user ?? NimbusUser()
+        user.configureGdprConsent(didConsent: true, consentString: consent)
+        NimbusRequestManager.user = user
+    }
+    
+    @objc public func destroyExistingAd() {
+        adController?.destroy()
+        adView?.removeFromSuperview()
+        adView = nil
+    }
+    
+}
+
+// MARK: - NimbusAdManagerDelegate implementation
+
+extension NimbusManager: NimbusAdManagerDelegate {
+    
+    public func didCompleteNimbusRequest(request: NimbusRequest, ad: NimbusAd) {
+        // do nothing
+    }
+    
+    public func didFailNimbusRequest(request: NimbusRequest, error: NimbusError) {
+        UnitySendMessage(kCallbackTarget, "OnError", error.localizedDescription);
+    }
+    
+    public func didRenderAd(request: NimbusRequest, ad: NimbusAd, controller: AdController) {
+        self.adController = controller
+        self.adController?.delegate = self
+        UnitySendMessage(kCallbackTarget, "OnAdRendered", "");
+    }
+    
+}
+
+// MARK: - AdControllerDelegate implementation
+
+extension NimbusManager: AdControllerDelegate {
+    
+    public func didReceiveNimbusEvent(controller: AdController, event: NimbusEvent) {
+        var method = "OnAdEvent", eventName = ""
+        switch event {
+        case .loaded:
+            eventName = "LOADED"
+        case .loadedCompanionAd(width: _, height: _):
```

Comment:
What is this case? This is currently not a video event, nor is it parodied on the Android side. Nor do I have a C# listener for it and I'm not certain that it's a valid event for Game Devs.

Reply:
No worries, it doesn't do anything in the Unity Side. @inderdhir can you explain what it means?

Reply:
Inder put this in as a callback for when a companion ad is loaded which I do not have in the Android SDK. Since we are currently changing the behavior of how we use companion ads, i want to revisit this and have some consensus between all of us on whether or not this event is needed since it will fire at at the end of all rewarded videos from now on once we start adding our own end cards to them.

Reply:
I changed the code to do not send these events for now, since unity is not expecting for them yet.